File: katas/6 kyu/unique-in-order/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def unique_in_order(iterable):
  x = []
  if len(iterable) > 0:
    x.append(iterable[0])
    for i in range(len(iterable)-1):
      if iterable[i] != iterable[i+1]:
        x.append(iterable[i+1])
  return x
# ------------------------------ CLEVER SOLUTION ----------------------------- #
unique_in_order = lambda l: [z for i, z in enumerate(l) if i == 0 or l[i - 1] != z]
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(unique_in_order('AAAABBBCCDAABBB'), ['A','B','C','D','A','B'])

File: katas/6 kyu/unique-in-order/solution.php
<!-- unique-in-order -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function uniqueInOrder($iterable){
  $arr = array();

  if (gettype($iterable) === "string") {
    if (strlen($iterable) < 2) return $iterable == "" ? [] : [$iterable];
    else $iterable = str_split($iterable);
  }

  array_push($arr,$iterable[0]);
  foreach (range(0 , count($iterable)-2)  as $i){
    if ($iterable[$i] != $iterable[$i+1]) array_push($arr , $iterable[$i+1]);
  }
  return $arr;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function uniqueInOrder($iterable){ 
  $arr = is_string($iterable) ? str_split($iterable) : $iterable;
  $ret = array_reduce($arr, function($carry, $item) {
    if ($item != end($carry)) {
      $carry[] = $item;
    }
    return $carry;
  }, []);
  
  return $ret;
}
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase
{
    // test function names should start with "test"
    public function testSampleTest() {
      $this->assertSame(['A','B','C','D','A','B'], uniqueInOrder('AAAABBBCCDAABBB'));
      $this->assertSame([], uniqueInOrder(''));
      $this->assertSame(['A'], uniqueInOrder('A'));
    }
}